### PR TITLE
Merge lite core changes

### PR DESCRIPTION
**Pull request by jayahariv**
partial merge to master (with latest lite core)

* includes API change in lite core, fleece etc..  release instead of free
* new replicator changes like, 
  * remove retry logic(move to lite core)
  * remove suspended state(move to lite core)
  * remove error handling(moved to lite core)
* disable some tests since, they are failing and not yet analyzed, will commit them via separate PR to master. 
* mbedtls, mbedcrypto, mbedx509 flags included.

### CODE REVIEW

File: Objective-C/CBLReplicator.mm [hunk 1]
```diff
@@ -148,11 +129,6 @@ - (NSString*) description {
     return _desc;
 }
 
-- (void) clearRepl {
-    c4repl_free(_repl);
-    _repl = nullptr;
-}
-
```
Comment (bmeike) [inline Objective-C/CBLReplicator.mm]: You can still tell that it has been closed/freed?

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: i am assuming, we are not deleting the c4replicator instance, and it should be handled by lite core.

Comment (bmeike) [reply Objective-C/CBLReplicator.mm]: 👍

File: Objective-C/CBLReplicator.mm [hunk 2]
```diff
@@ -713,33 +607,13 @@ - (BOOL) active {
...
-                return;
-            }
-            
-            if (suspended) {
-                _state = kCBLStateSuspending;
-                [self _stop];
-            } else
-                [self resetAndScheduleRetry];
-        }
+        c4repl_setSuspended(_repl, suspended);
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: @jayahariv Have you looked if the logic to [1] ignore the suspend when the replicator is in the stop process or [2] to resume from suspending (e.g. switching fast b/w BG and FG) is still working?

At least we should create a ticket to verify after all other unit tests are passed.

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: sure, will create a ticket to verify this

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: https://issues.couchbase.com/browse/CBL-687

**Pull request merged**